### ISSUE TITLE
Better way of distributing previews?

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#26741]
I like being on the cutting edge of .Net technologies, so I often use VS previews and .Net Core previews ("official" ones and even daily builds).

But I find using Roslyn previews too hard, so I don't bother. Some issues I have with the way they are currently set up:

1. There are two previews ([vNext Preview](https://github.com/dotnet/csharplang/wiki/vNext-Preview) and [Nullable Reference Types Preview](https://github.com/dotnet/csharplang/wiki/Nullable-Reference-Types-Preview)), presumably mutually incompatible.
2. The instructions say I can't update VS while the preview is installed. This means that if I want to keep the preview installed while keeping VS updated, I have to keep installing and uninstalling it.
3. Having to extract an archive and run a bat file is inconvenient.

If you want to make sure more people use the previews, I think you should make it easier. A great solution would be something like an experimental option that can be toggled in VS Preview settings.

**Comment by jcouv** [comment]
@svick Yes, we definitely recognize the limitations of the current distribution mechanism. We are actively investigating other technical solutions, but haven't found a great one at the moment.

As for keeping the nullable and patterns+range preview separate, that was a policy/cost decision. The nullable codebase has diverged a lot and is not trivial to merge with other features. We'll merge it once it is more complete.

Tagging @cston @jaredpar @jmarolf FYI

**Comment by CyrusNajmabadi** [comment]
Closing as stale.
